app/constructor/Enemy.js

```diff
@@ -109,7 +113,24 @@ export default class Enemy extends entityPrefab {
       this.animations.play(this.orientation, 30, true)
     }
   }
-
+  takeDamage(damage) {
+    console.log('damage taken is:', damage, 'armor:', this.stats.defense)
+    this.stats.hp -= (damage - this.stats.defense)
+    console.log('health reduced to:', this.stats.hp)
+    //  check if dead
+    if (this.stats.hp <= 0) {
+      this.die()
+      //  function returns true if the enemy is dead
+      return true
+    }
+    //  returns false because the enemy didn't die
+    return false
+  }
+  attack() {
+    if (Date.now() - this.lastAttack < 900) return 0
```

Comment:
@gabrielwr Why are we using the date in the attack function. Is it to keep track of when monster attacks?